src/Microsoft.Health.Fhir.Liquid.Converter/Hl7v2/Hl7v2Processor.cs

```diff
@@ -57,57 +52,17 @@ public string Convert(string data, string rootTemplate, ITemplateProvider templa
             return result.ToString(Formatting.Indented);
         }
 
-        public string Convert(string data, string rootTemplate, ITemplateProvider templateProvider, CancellationToken cancellationToken, TraceInfo traceInfo = null)
+        protected override Context CreateContext(ITemplateProvider templateProvider, object hl7v2Data)
         {
-            cancellationToken.ThrowIfCancellationRequested();
-            return Convert(data, rootTemplate, templateProvider, traceInfo);
-        }
-
-        private Context CreateContext(ITemplateProvider templateProvider, Hl7v2Data hl7v2Data)
-        {
-            // Load data and templates
-            var timeout = _settings?.TimeOut ?? 0;
-            var context = new Context(
-                environments: new List<Hash>() { Hash.FromDictionary(new Dictionary<string, object>() { { "hl7v2Data", hl7v2Data } }) },
-                outerScope: new Hash(),
-                registers: Hash.FromDictionary(new Dictionary<string, object>() { { "file_system", templateProvider.GetTemplateFileSystem() } }),
-                errorsOutputMode: ErrorsOutputMode.Rethrow,
-                maxIterations: 0,
-                timeout: timeout,
-                formatProvider: CultureInfo.InvariantCulture);
-
             // Load code system mapping
+            var context = base.CreateContext(templateProvider, hl7v2Data);
             var codeSystemMapping = templateProvider.GetTemplate("CodeSystem/CodeSystem");
             if (codeSystemMapping?.Root?.NodeList?.First() != null)
```

Comment:
can we move "codeSystemMapping?.Root?.NodeList?.First()"  into GetTemplate, which might be more clear?

Reply:
Got it. Currently `GetTemplate` returns `Template`. If we want to return a mapping we may need to change the return type from `Template` to `object`, or create another `GetMapping` interface in `ITemplateProvider` (but maybe not all template provider will have such a mapping?). Do you have any suggestions about it?